File: solutions/lua/resistor-color-duo/1/resistor-color-duo.lua
return {
    value = function(colors)
        local rcc = {
            ['black' ] = 0,
            ['brown' ] = 1,
            ['red'   ] = 2,
            ['orange'] = 3,
            ['yellow'] = 4,
            ['green' ] = 5,
            ['blue'  ] = 6,
            ['violet'] = 7,
            ['grey'  ] = 8,
            ['white' ] = 9,
        }
        return rcc[colors[1]] * 10 + rcc[colors[2]]
    end
}

File: solutions/php/resistor-color-duo/1/ResistorColorDuo.php
<?php

declare(strict_types=1);

class ResistorColorDuo
{
    public function getColorsValue(array $colors): int
    {
        $band_colors = array(
            'black'  => 0,
            'brown'  => 1,
            'red'    => 2,
            'orange' => 3,
            'yellow' => 4,
            'green'  => 5,
            'blue'   => 6,
            'violet' => 7,
            'grey'   => 8,
            'white'  => 9,
        );
        $color1 = $band_colors[$colors[0]];
        $color2 = $band_colors[$colors[1]];
        return $color1 * 10 + $color2;
    }
}
